### PR TITLE
Deactivate Django users not found in LDAP

### PR DESCRIPTION
**Pull request by busla**

### CODE REVIEW

**Comment by etianen** [comment]
Hi,

Thanks for the pull request!

It feels like this is a bit of a special case, so I'm not sure it's worth the long-term maintenance of adding it to django-python3-ldap.

However, if you feel strongly that other people will find it useful, and are happy to maintain it yourself, then go ahead and make it a reusable Django app that depends on django-python3-ldap. I'll then link to it from the project documentation, making it easy for other people to find and benefit from your work.

**Comment by busla** [comment]
Ok, sounds good, thanks.

**Comment by etianen** [comment]
Let me know when your project goes online, and I'll update the docs. :)

**Comment by frennkie** [comment]
I would also find this useful. Having a separate App and Project seems like a lot of overhead for one additional management command.

@etianen could you maybe reconsider this? :-)

If not I'll simply add the code directly into my app...

**Comment by etianen** [comment]
Okay, I'll reconsider this feature, due to popular (2 users!) demand.

I would suggest refactoring this so that it's not a separate command. A `--deactivate` flag for ldap_sync_users would seem to be the ticket here. What do people think?

**Comment by busla** [comment]
Sounds great! Thanks for reconsidering.

I agree, a separate custom command is reduntant, since 90% of that code is a taken from `ldap_sync_users`. It was a quick fix.

**Comment by WilliamZimmermann** [comment]
Hello! This function is useful for me too. Have you a prevision about when this pull request will be marged?

**Comment by etianen** [comment]
I'm still waiting for the changes in this comment:

https://github.com/etianen/django-python3-ldap/pull/80#issuecomment-372275112

I'm happy for someone to take over this pull request and make the changes. Alternatively, I'll do it myself when I have time.

File: django_python3_ldap/management/commands/ldap_sync_and_deactivate_users.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from django_python3_ldap import ldap
+from django_python3_ldap.conf import settings
+from django.contrib.auth.models import User
```
Comment (WilliamZimmermann) [inline django_python3_ldap/management/commands/ldap_sync_and_deactivate_users.py]: I tried to run this scrupt but, after read all activate users and put them into the list (line 29) and take the difference (line 31), it shows a error: Manager isn't available...

I solved it by change this line (6) to:
```
from django.contrib.auth import get_user_model
User = get_user_model()
```

Comment (busla) [reply django_python3_ldap/management/commands/ldap_sync_and_deactivate_users.py]: thanks! ofcourse there needs to be support for a custom User model.

fixed in: https://github.com/busla/django-python3-ldap/commit/4ae7ab3f3765bf0e4b9ea80377964c1674e37de0

**Comment by alex-dot** [comment]
Just found this. This would be a great boon to us. So 👍

**Pull request closed without merging**